.github/workflows/main.yml

```
@@ -32,7 +32,7 @@ jobs:
               runner: "ubuntu-latest"
             stack-yaml: "stack.ghc-8.10.yaml"
       fail-fast: false
-    name: ${{ matrix.os.runner }} - ${{ matrix.stack-yaml }}
+    name: ${{ matrix.os.runner }} - ${{matrix.os.runner.arch}} - ${{ matrix.stack-yaml }}
```

According to the docs that should be the only key where we cannot use the `runner` context, but that is ok.
```suggestion
    name: ${{ matrix.os.runner }} - ${{ matrix.stack-yaml }}
```